### PR TITLE
Migrate Semester. Rearrange syntax/parser/arb.

### PR DESCRIPTION
**Pull request by rpiaggio**
* Migrate  from `gsp-core`: `Semester`, specific enum implicits and parsers.
* Move some `syntax`, `parser`, `arb` implementations out of `math` package and under `model`, `optics` or root.
* `import lucuma.core.syntax.all._` is available.
* Use `syntax.time` when possible. Added `ZonedDateTime` `+`/`-`.

Some formatting.

### CODE REVIEW

File: modules/core/shared/src/main/scala/lucuma/core/enum/package.scala [hunk 1]
```diff
@@ -0,0 +1,85 @@
+// Copyright (c) 2016-2020 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package lucuma.core
+
+import java.time.Month
+import java.time.Month._
+import cats.implicits._
```
Comment (swalker2m) [inline modules/core/shared/src/main/scala/lucuma/core/enum/package.scala:8]: Will this get the cats updates, (e.g. to `import cats.syntax.all._`)?

File: modules/core/shared/src/main/scala/lucuma/core/enum/package.scala [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+import lucuma.core.optics.syntax.ToPrismOps
+
+// The members of this package are generated from database tables, which are the source of truth.
+// See project/gen2.scala for details. Associations with other model types, as needed, are provided
```
Comment (swalker2m) [inline modules/core/shared/src/main/scala/lucuma/core/enum/package.scala:12]: We can delete the first two sentences here since they no longer apply.

**Review by swalker2m — APPROVED** [review]
Looks great. :+1:

File: modules/core/shared/src/main/scala/lucuma/core/enum/package.scala (hunk 1 above)
Comment (rpiaggio) [reply modules/core/shared/src/main/scala/lucuma/core/enum/package.scala:8]: Yup, slipped through the cracks. Sending another commit.

File: modules/core/shared/src/main/scala/lucuma/core/enum/package.scala (hunk 2 above)
Comment (rpiaggio) [reply modules/core/shared/src/main/scala/lucuma/core/enum/package.scala:12]: :+1:

Comment (rpiaggio) [reply modules/core/shared/src/main/scala/lucuma/core/enum/package.scala:12]: Hmm... Actually we could move the extra methods to the enums themselves then. I'll do that.

Comment (swalker2m) [reply modules/core/shared/src/main/scala/lucuma/core/enum/package.scala:12]: Oh, right, that makes more sense now.

**Review by swalker2m — APPROVED** [review]
Looks good.  :+1:

**Pull request merged**